From bernhard.rosenkraenzer.ext@huawei.com Tue Jul 13 04:09:16 2021
From: Bernhard Rosenkraenzer <bernhard.rosenkraenzer.ext@huawei.com>
Date: Tue, 13 Jul 2021 04:09:16 +0100
Subject: [PATCH] Fix build with musl 1.2.2

alloc-util.h redefines reallocarray() if HAVE_REALLOCARRAY isn't set,
but there's nothing that sets HAVE_REALLOCARRAY -- so reallocarray
is redefined unconditionally. The define seems to be a leftover from
the time when systemd was using autoconf.

Since reallocarray's prototype differs slightly in some libcs (such as
musl 1.2.2), this breaks the build.

Signed-off-by: Bernhard Rosenkraenzer <bernhard.rosenkraenzer.ext@huawei.com>
Upstream-Status: Pending

diff -up git/src/basic/alloc-util.h.omv~ git/src/basic/alloc-util.h
--- git/src/basic/alloc-util.h.omv~	2021-07-10 21:19:45.461902769 +0200
+++ git/src/basic/alloc-util.h	2021-07-10 21:21:43.505118481 +0200
@@ -96,7 +96,7 @@ _malloc_  _alloc_(1, 2) static inline vo
         return malloc(size * need ?: 1);
 }
 
-#if !HAVE_REALLOCARRAY
+#if 0 /* Every libc we support has reallocarray, and prototypes differ slightly */
 _alloc_(2, 3) static inline void *reallocarray(void *p, size_t need, size_t size) {
         if (size_multiply_overflow(size, need))
                 return NULL;
